src/main.rs

```diff
@@ -32,28 +32,30 @@ fn generate_short_id() -> String {
         .collect()
 }
 
-/// Get passcode from request's cookie
+// Get passcode from request's cookie
```

Comment:
```
fn get_req_passcode(req: &Request) -> Result<String> {
    let cookie_val: &str = req
        .get_header(header::COOKIE)
        .ok_or_else(|| anyhow!("No cookie found"))?
        .to_str()?;

    // we split at ";" not "; ", in case the cookie is ending with ";"
    cookie_val
    .split(';')
    .find_map(|kv| {
        let index = kv.find('=')?;
        let (key, value) = kv.split_at(index);
        if key.trim() != "passcode" {
            return None;
        }

        // remove the "="
        let value = value.trim_start_matches('=').to_string();
        Some(value)
    })
    .ok_or_else(|| anyhow!("No passcode found in cookie"))
}
```
I think previous version with your function `trim_start_matches` is better, because: 
- less verbose
- more performant,  no extra Result -> Option converstion during iteration